system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -98,23 +166,19 @@ class SubscriberTopicStatisticsNode : public system_metrics_collector::MetricsMe
     publisher_;
 
   /**
-  * ROS 2 topic statistics collectors
-  */
-  std::vector<std::shared_ptr<TopicStatisticsCollector<T>>> statistics_collectors_;
+   * ROS2 topic statistics collectors
+   */
+  std::vector<std::unique_ptr<TopicStatisticsCollector<T>>> statistics_collectors_;
 
 private:
   /**
    * Creates ROS2 timers and a publisher for periodically triggering measurements
    * and publishing MetricsMessages
-   *
-   * @return if setup was successful
    */
   void StartPublisher();
 
   /**
    * Stops the ROS2 timers that were created by StartPublisher()
-   *
-   * @return if teardown was successful
    */
   void StopPublisher();
```

Comment:
It looks like the .cpp file was removed. Is this still needed given the definition below?

Reply:
You mean separating the declaration and definition? The way this class is structured (all declarations followed by all definitions) makes it more readable.

Reply:
Is there a convention for that? We didn't follow this convention for the topic statistics declarations, so I mention this to be consistent.

Reply:
Discussed offline - making it consistent to topic statistics classes.